src/dynarec/rv64/dynarec_rv64_functions.c

```diff
@@ -764,3 +765,21 @@ int fpu_is_st_freed(dynarec_rv64_t* dyn, int ninst, int st)
 {
     return (dyn->e.tags&(0b11<<(st*2)))?1:0;
 }
+
+void updateNativeFlags(dynarec_rv64_t* dyn)
+{
+    if (!box64_dynarec_nativeflags)
+        return;
+    for (int i = 1; i < dyn->size; ++i)
+        if (dyn->insts[i].nat_flags_fusion) {
+            if (dyn->insts[i].pred_sz == 1 && dyn->insts[i].pred[0] == i - 1
+                && (dyn->insts[i].x64.use_flags & dyn->insts[i - 1].x64.set_flags) == dyn->insts[i].x64.use_flags) {
+                dyn->insts[i - 1].nat_flags_fusion = 1;
+                if (dyn->insts[i].x64.use_flags & X_SF) {
+                    dyn->insts[i - 1].nat_flags_needsign = 1;
+                }
+                dyn->insts[i].x64.use_flags = 0;
```

Comment:
I still think doing this handling before ̀`updateNeeds(...)` will break some case were a flags is used in 2 opcodes.

Reply:
I guess what bother me is that the "use_flag" information get overwritten.
Anyway, the potential issue is "tranfomation" flag maybe? While unlikly a jump can go to a conditional operation that needs "native flags".